package/MDAnalysis/lib/formats/xtc.pyx

```diff
@@ -0,0 +1,309 @@
+        return True
+
+    def open(self, fname, mode):
+        """Open a XTC file
+
+        If another XTC file is currently opened it will be closed
+
+        Parameters
+        ----------
+        fname: str
+            The filename to open.
+        mode: ('r', 'w')
+            The mode in which to open the file, either 'r' read or 'w' write
+
+        Raises
+        ------
+        ValueError
+            Wrong mode given
+        IOError
+            Couldn't read the file
+        """
+        if self.is_open:
+            self.close()
+        cdef int return_code = 1
+        self.fname = fname
+        self.n_atoms = 0
+        self.reached_eof = False
+        self.current_frame = 0
+
+        if mode == 'r':
+            opening_mode = b'r'
+        elif mode == 'w':
+            opening_mode = b'w'
+        else:
+            raise ValueError('mode must be one of "r" or "w", you '
+                             'supplied {}'.format(mode))
+        self.mode = mode
+
+        if self.mode == 'r':
+            if not exists(self.fname):
+                raise IOError('File does not exists: {}'.format(self.fname))
+
+            return_code = xdrlib.read_xtc_natoms(fname, &self.n_atoms)
+
+            if return_code != 0:
+
+                raise IOError('XTC read error: {}'.format(return_code))
+            if self.n_atoms <= 0:
+                raise IOError('Couldn\'t read number of atoms: {}'.format(
+                    self.fname))
+
+        self.xfp = xdrlib.xdrfile_open(fname, opening_mode)
+        if self.xfp is NULL:
+            raise IOError('error opening xtf file: {}'.format(self.fname))
+        self.is_open = True
+
+    def __enter__(self):
+        """Support context manager"""
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Support context manager"""
+        self.close()
+        # always propagate exceptions forward
+        return False
+
+    def __iter__(self):
+        self.close()
+        self.open(self.fname, self.mode)
+        return self
+
+    def __next__(self):
+        if self.reached_eof:
+            raise StopIteration
+        return self.read()
+
+    # TODO: cache result in file
+    def __len__(self):
+        if not self.is_open:
+            raise RuntimeError('No file currently opened')
+        return self.offsets.size
+
+    def seek(self, frame):
+        """Seek to Frame
+
+        Parameters
+        ----------
+        frame: int
+            wind the file to given frame
+
+        Raises
+        ------
+        RuntimeError
+            If you seek for more frames then are available
+        """
+        if frame >= self.offsets.size:
+            raise RuntimeError('Trying to seek over max number of frames')
+        print(frame, self.offsets[frame])
+        # print(xdrlib.xdr_seek(self.xfp, self.offsets[frame], xdrlib.SEEK_SET))
+        print(xdrlib.xdr_seek(self.xfp, 0, xdrlib.SEEK_SET))
```

Comment:
Yea... I couldn't also tell it would be the cause, but do flag it as problematic because `exdrNR` shouldn't be used for an error code and it will break when we attempt to retrieve the error message from it.
Anyway, a Google search led me to http://pythonextensionpatterns.readthedocs.org/en/latest/exceptions.html, where it is pointed out that somewhere a function is returning `NULL` without setting an exception. None of the seek functions can return a `NULL`...